// cli-itu/lib/utils/gh.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.isAuthenticated = exports.isInstalled = void 0;
const shell = require("shelljs");
const child_process_1 = require("child_process");
// Credit: https://github.com/Kvalifik/Kvalifik-CLI/blob/development/src/utils/gh.ts
const isInstalled = () => {
    const version = shell.exec("gh --version", { async: false, silent: true });
    if (!version.includes("gh version")) {
        console.error("❌ Github CLI is not installed. Please install and try again");
        console.info("For instructions, see https://github.com/cli/cli#installation");
        return false;
    }
    return true;
};
exports.isInstalled = isInstalled;
const isAuthenticated = () => {
    const authStatus = shell.exec("gh auth status", {
        async: false,
        silent: true,
    });
    if (authStatus.code === 1) {
        console.info("❌ Not logged in to Github.");
        console.info("Please follow the next steps to authenticate with Github, and then re-run the command to continue");
        (0, child_process_1.spawn)("gh auth login", ["--web"], {
            stdio: "inherit",
            shell: true,
        });
        return false;
    }
    return true;
};
exports.isAuthenticated = isAuthenticated;
