// cli-itu/lib/commands/open.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const command_1 = require("@oclif/command");
const cli_ux_1 = require("cli-ux");
class Open extends command_1.Command {
    constructor() {
        super(...arguments);
        this.platforms = {
            learnit: "https://learnit.itu.dk/",
            student: "https://itustudent.itu.dk/",
            timeedit: "https://cloud.timeedit.net/itu/web/public/",
            mystudyactivities: "https://minestudieaktiviteter.itu.dk/",
            itu: "https://itu.dk",
            github: "https://github.itu.dk/login",
            kattis: "https://itu.kattis.com/",
        };
    }
    async run() {
        const { args } = this.parse(Open);
        const url = this.platforms[args.Platform];
        this.log(`Opening ${url} in default web browser`);
        await cli_ux_1.default.open(url);
    }
    async catch(error) {
        throw error;
    }
}
exports.default = Open;
Open.aliases = ["o"];
Open.description = "open an ITU platform";
Open.examples = ["$ itu open learnit", "$ itu open timeedit"];
Open.flags = {
    help: command_1.flags.help({ char: "h" }),
};
Open.args = [
    {
        name: "Platform",
        required: true,
        description: "ITU platform to open",
        default: "learnit",
        options: [
            "learnit",
            "student",
            "timeedit",
            "mystudyactivities",
            "itu",
            "github",
            "kattis",
        ],
    },
];
